components/slack/slack.app.mjs

```diff
@@ -450,29 +472,48 @@ export default {
      * token
      */
     sdk() {
-      return new WebClient(this.getToken());
+      return new WebClient(this.getToken(), {
+        rejectRateLimitedCalls: true,
+      });
     },
     async makeRequest({
-      method = "", ...args
+      method = "", throwRateLimitError = false, ...args
     } = {}) {
-      let response;
       const props = method.split(".");
       const sdk = props.reduce((reduction, prop) =>
         reduction[prop], this.sdk());
 
-      try {
-        response = await sdk(args);
-      } catch (error) {
-        console.log(`Error calling ${method}`, error);
-        throw error;
-      }
+      const response = await this._withRetries(() => sdk(args, throwRateLimitError));
 
       if (!response.ok) {
         console.log(`Error in response with method ${method}`, response.error);
         throw response.error;
       }
       return response;
     },
+    async _withRetries(apiCall, throwRateLimitError = false) {
+      const retryOpts = {
+        retries: 3,
+        factor: 2,
+        minTimeout: 3000,
+      };
+      return retry(async (bail) => {
+        try {
+          return await apiCall();
+        } catch (error) {
+          const statusCode = get(error, "code");
+          if (statusCode === 429) {
+            if (throwRateLimitError) {
+              bail(`Rate limit exceeded. ${error}`);
```

Comment:
Do you know if this message will be shown in the async options UI?

Reply:
Hi @GTFalcao. I did some testing, made some minor adjustments, and yes the message is displayed in the async options!
<img width="632" alt="Screenshot 2025-03-26 at 3 13 54 PM" src="https://github.com/user-attachments/assets/f1c72370-7d81-4905-b02d-59dae43f09ff" />